### PR TITLE
Image Analysis SDK updates for beta.2 release - Part 2

### PR DESCRIPTION
**Pull request by dargilco**
A few fixes related to Part 1 that did not make it into the [Part 1 PR](https://github.com/Azure/azure-sdk-for-java/pull/38694):
* Related to the switch from java.net.URL to String for input Image URL:
  * Remove unused `import java.net.MalformedURLException;`
  * Remove unneeded `throws MalformedURLException` 
  * Add a breaking change note about this in CHANGELOG.md (thank you Srikanta!)

Additional changes, per recent arch-board review:
  * Add new public operation methods that support input `RequestOptions` and returns `Response<ImageAnalysisResult>`. 
    * Add two new samples.
    * Add four new tests. Verify test recordings and load them into Azure/azure-sdk-assets repo.
    * Update root and samples README.md docs. 
 
Other:
* Update CHANGELOG.md
* Fix a few typos in source code comments
* Update samples to use class-level JavaDoc style comment for what the sample does

### CODE REVIEW

**Review by srnagar — APPROVED** [review]

File: sdk/vision/azure-ai-vision-imageanalysis/CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,24 @@
 
 ### Breaking Changes
 
-- In the previous version, you would call the `analyze` method on the `ImageAnalysisClient` or `ImageAnalysisAsyncClient` to analyze an image from a publicly accessible URL, or from a memory buffer. To better align with other Azure client libraires, this was changed in this release. Call the new dedicated `analyzeFromUrl` method to analyze an image from URL. Keep calling the `analyze` method to analyze an image from a memory buffer.
+- In the previous version, you would call the `analyze` method on the `ImageAnalysisClient` or `ImageAnalysisAsyncClient` to analyze an image from a publicly accessible URL, or from a memory buffer. To better align with other Azure client libraires, this was changed in this release. Call the new dedicated `analyzeFromUrl` method to analyze an image from URL (and note that you now proive the URL as a `String` instead of `java.net.URL`). Keep calling the `analyze` method to analyze an image from a memory buffer.
```
[the remark this answers is not in the document]
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/CHANGELOG.md]: Fixed, thanks you!

File: sdk/vision/azure-ai-vision-imageanalysis/src/main/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClient.java [hunk 2]
```diff
@@ -558,6 +559,50 @@ ImageAnalysisResult analyzeFromImageData(List<VisualFeatures> visualFeatures, Bi
...
+     * Apply the Image Analysis Options as query parameters on a given RequestOptions object.
+     * Return an updated RequestOptions object.
+     *
+     * @param inputRequestOptions The input request options (can be null).
+     * @param imageAnalysisOptions The Image Analysis Options to apply to the request to be sent (can be null).
+     *
+     * @return An updated RequestOptions with addition query parameters set based on the Image Analysis Options. Can be
+     * null.
+     */
+    static RequestOptions updateRequestOptions(RequestOptions inputRequestOptions,
```
[the remark this answers is not in the document]
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/main/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClient.java:572]: I removed the duplicate method, and made the code in ImageAnalysisAsyncClient.java call the method in ImageAnalysisClient.java. Thanks!!

File: sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/SampleCaptionImageFileWithResponse.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+//
+//     This sample is similar to the `SampleCaptionImageFile.java` sample, but it uses the 
+//     `analyzeWithResponse` method instead of `analyze` method. This allows customization of the 
+//     HTTP request, and access to HTTP request and response details. This is not commonly required, but can
+//     be useful for service customization or troubleshooting issues.
+//
+//     Set these two environment variables before running the sample:
+//     1) VISION_ENDPOINT - Your endpoint URL, in the form https://your-resource-name.cognitiveservices.azure.com
+//                          where `your-resource-name` is your unique Azure Computer Vision resource name.
+//     2) VISION_KEY - Your Computer Vision key (a 32-character Hexadecimal number)
```
[the remark this answers is not in the document]
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/SampleCaptionImageFileWithResponse.java]: Done!

File: sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/SampleOcrImageUrl.java [hunk 4]
```diff
@@ -2,7 +2,7 @@
 // Licensed under the MIT License.
 //
 // DESCRIPTION:
-//     This sample demonstrates how to extract printed or hand-written text for of a publicly accessible
+//     This sample demonstrates how to extract printed or hand-written text from a publicly accessible
```
[the remark this answers is not in the document]
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/SampleOcrImageUrl.java]: Done!

**Review by srnagar — APPROVED** [review]
LGTM!

**Pull request merged**